### PR TITLE
Add lr_scheduler, weight decay and max grad norm to finetune object

### PR DESCRIPTION
**Pull request by artek0chumak**
Issue: ENG-13459

Screenshot 2024-11-19 at 12 33 40

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -408,11 +408,25 @@ paths:
                   format: float
                   default: 0.00001
                   description: Learning rate multiplier to use for training
+                lr_scheduler:
+                  type: object
+                  default: none
+                  $ref: '#/components/schemas/LRSceduler'
```
Comment (mryab) [inline openapi.yaml]:
```suggestion
                  $ref: '#/components/schemas/LRScheduler'
```

File: openapi.yaml [hunk 2]
```diff
@@ -408,11 +408,25 @@ paths:
...
                 warmup_ratio:
                   type: number
                   format: float
                   default: 0.0
                   description: The percent of steps at the start of training to linearly increase the learning rate.
+                max_grad_norm:
+                  type: number
+                  format: float
+                  default: 1.0
+                  description: Max gradient norm to be used for gradient clipping.
```
Comment (mryab) [inline openapi.yaml:424]:
```suggestion
                  description: Max gradient norm to be used for gradient clipping. Set to 0 to disable.
```

File: openapi.yaml [hunk 3]
```diff
@@ -2039,3 +2062,21 @@ components:
         - type
         - lora_r
         - lora_alpha
+    LRSceduler:
```
Comment (mryab) [inline openapi.yaml]:
```suggestion
    LRScheduler:
```

**Review by mryab — APPROVED** [review]

**Review by punkerpunker — APPROVED** [review]
